proto/task_execution.proto

```diff
@@ -23,6 +23,11 @@ message TaskParameter {
   //data is a directory, if used for an output all the files in the directory
   //will be copied to the storage location
   bool   directory = 5;
+  //if the parameter is an output, should the element be created before executing
```

Comment:
Thanks for clarifying.  My Spidey sense is telling me that folks might misinterpret `path`.  Maybe in a future PR possibly re-naming it to `containerMappedPath`, or just `containerPath` might be more explicit.  The location could be named as `cloudPath` or something like that, with directory renamed as `arePathsDirectories` - though many times an API can tell if a path is a directory or a file.

On a separate nit, some of the documentation for the fields seem like phrases and start with lower-case letters.  As this will become a standard that others will try to implement, it might be nice to have them as "proper" as possible.  For example, I had to read this sentence about five times before it clicked:

>//if the parameter is an output, should the element be created before executing
//the command.